src/ds/d500/d500-auto-calibration.cpp

```diff
@@ -26,57 +26,25 @@ namespace librealsense
         "Failed to Run"
     };
 
-#pragma pack(push, 1)
-    struct d500_calibration_answer
-    {
-        uint8_t calibration_state;
-        int8_t calibration_progress;
-        uint8_t calibration_result;
-        ds::d500_coefficients_table depth_calibration;
-    };
-#pragma pack(pop)
-
-
     d500_auto_calibrated::d500_auto_calibrated() :
         _mode (d500_calibration_mode::RS2_D500_CALIBRATION_MODE_RESERVED),
         _state (d500_calibration_state::RS2_D500_CALIBRATION_STATE_IDLE),
         _result(d500_calibration_result::RS2_D500_CALIBRATION_RESULT_UNKNOWN)
-    {}
-
-    bool d500_auto_calibrated::check_buffer_size_from_get_calib_status(std::vector<uint8_t> res) const
     {
-        // the GET_CALIB_STATUS command will return:
-        // - 3 bytes during the whole process
-        // - 515 bytes (3 bytes + 512 bytes of the depth calibration) when the state is Complete
-
-        bool is_size_ok = false;
-        if (res.size() > 1)
-        {
-            if (res[0] < static_cast<int>(d500_calibration_state::RS2_D500_CALIBRATION_STATE_COMPLETE) &&
-                res.size() == (sizeof(d500_calibration_answer) - sizeof(ds::d500_coefficients_table)))
-                is_size_ok = true;
-
-            if (res[0] == static_cast<int>(d500_calibration_state::RS2_D500_CALIBRATION_STATE_COMPLETE) &&
-                res.size() == sizeof(d500_calibration_answer))
-                is_size_ok = true;
-        }
-        return is_size_ok;
+        // add here choice between hw_monitor or debug protocol
+        // is hw monitor needed at all (since debug protocol uses hwm
+        _ac_handler = std::make_shared <d500_auto_calibrated_handler>();
```

Comment:
If we wanted to replace it with a DDS implementation, then it would be a different class, and it would get a `dds_device` rather than a `debug_protocol`, right?
Why not have the owner send the implementation in, with the pointer already set so you wouldn't have to have `set_device_for_auto_calib`?

Reply:
done